dsp/window/window_test.go

```diff
@@ -5,82 +5,111 @@
 package window
 
 import (
+	"fmt"
 	"testing"
 
 	"gonum.org/v1/gonum/floats"
 )
 
 var windowTests = []struct {
-	name   string
-	fn     func([]float64) []float64
-	want   []float64
-	winLen int
+	name    string
+	fn      func([]float64) []float64
+	fnCmplx func([]complex128) []complex128
+	winLen  int
+	want    []float64
 }{
 	{
-		name: "Rectangular", fn: Rectangular, winLen: 20,
-		want: []float64{1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-			1, 1, 1, 1, 1, 1, 1, 1, 1, 1},
+		name: "Rectangular", fn: Rectangular, fnCmplx: RectangularComplex, winLen: 20,
+		want: []float64{
+			1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+			1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
+		},
 	},
 	{
-		name: "Sine", fn: Sine, winLen: 20,
-		want: []float64{0.000000, 0.164595, 0.324699, 0.475947, 0.614213, 0.735724, 0.837166, 0.915773, 0.969400, 0.996584,
-			0.996584, 0.969400, 0.915773, 0.837166, 0.735724, 0.614213, 0.475947, 0.324699, 0.164595, 0.000000},
+		name: "Sine", fn: Sine, fnCmplx: SineComplex, winLen: 20,
+		want: []float64{
+			0.078459, 0.233445, 0.382683, 0.522499, 0.649448, 0.760406, 0.852640, 0.923880, 0.972370, 0.996917,
+			0.996917, 0.972370, 0.923880, 0.852640, 0.760406, 0.649448, 0.522499, 0.382683, 0.233445, 0.078459,
+		},
 	},
 	{
-		name: "Lanczos", fn: Lanczos, winLen: 20,
-		want: []float64{0.000000, 0.115514, 0.247646, 0.389468, 0.532984, 0.669692, 0.791213, 0.889915, 0.959492, 0.995450,
-			0.995450, 0.959492, 0.889915, 0.791213, 0.669692, 0.532984, 0.389468, 0.247646, 0.115514, 0.000000},
+		name: "Lanczos", fn: Lanczos, fnCmplx: LanczosComplex, winLen: 20,
+		want: []float64{
+			0.052415, 0.170011, 0.300105, 0.436333, 0.57162, 0.698647, 0.810332, 0.900316, 0.963398, 0.995893,
+			0.995893, 0.963398, 0.900316, 0.810332, 0.698647, 0.57162, 0.436333, 0.300105, 0.170011, 0.052415,
+		},
 	},
 	// This case tests Lanczos for a NaN condition. The Lanczos NaN condition is k=(N-1)/2, that is when N is odd.
```

Comment:
// This case tests Lanczos for a NaN condition. The Lanczos NaN condition is k=(N-2)/2, 
	// that is when N is even and N>=2.
So, test with winLen=21 not checks NaN condition now.